src/generic/UnivPoly.jl

```diff
@@ -1013,21 +1028,40 @@ function add!(a::UnivPoly{T}, b::UnivPoly{T}, c::UnivPoly{T}) where {T <: RingEl
    return a
 end
 
+function add!(a::UnivPoly{T}, b::UnivPoly{T}, c::Integer) where {T <: RingElement}
```

Comment:
What I really need is this but for `ZZRingElem` instead of `Integer`.

Come to think of, it also would make a lot of sense to have this for `c::T`.

And easy way to cover all those would be to just do
```suggestion
function add!(a::UnivPoly{T}, b::UnivPoly{T}, c) where {T <: RingElement}
```
There shouldn't be too much issues with ambiguity as long as two of the arguments have identical type.

But I still wonder if such a signature is really a good idea?

Reply:
I've now changed it to this (i.e. dropped the restriction on one of the arguments completely). 

Wondering if @fieker or @thofma forsee any potential problems with this?

(We would like these functions for GenericCharacterTables.jl so we can speed up things there w/o relying on implementation details for `UnivPoly`)

Reply:
no objection from me